src/utils.ts

```diff
@@ -13,3 +16,64 @@ export const aestheticBaseAttributes: Pick<Nft, 'aestheticRank' | 'aestheticRank
     timesWon: 0,
     aestheticRankReliability: 0,
 };
+
+/**
+ * A mapping of all PopRank supported chains to all
+ * of our supported marketplaces.
+ *
+ * The first marketplace under a chain is considered the default.
+ */
+export const chainMarketplaceMap: ChainMarketplaceMap = {
+    ethereum: {
+        opensea: {
+            name: "OpenSea",
+            nftUrlFormatter: (address?: string, id?: string) => `${OPENSEA_BASE_NFTS_URL}/${address}/${id}`,
+            collectionUrlFormatter: (address: string) => `${OPENSEA_BASE_COLLECTIONS_URL}/${address}`
+        }
+    },
+    polygon: {
+        opensea: {
+            name: "OpenSea",
+            nftUrlFormatter: (address?: string, id?: string) => `${OPENSEA_BASE_NFTS_URL}/matic/${address}/${id}`,
+            collectionUrlFormatter: (address: string) => `${OPENSEA_BASE_COLLECTIONS_URL}/${address}`
+        }
+    },
+    solana: {
+        magiceden: {
+            name: "MagicEden",
+            nftUrlFormatter: (_address?: string, id?: string) => `${MAGICEDEN_BASE_NFTS_URL}/${id}`,
```

Comment:
hmmm the first param being ignored here is kinda awkward, should we widen the type for `nftUrlFormatter` to improve the usage of this function?

Reply:
can you expand on what you mean by "widen the type"? Some sort of union of types?

The downside to that is that then on the frontend we're gonna have to have checks around what the `chain` is before calling the `nftUrlFormatter`, whereas if they all share the same arguments, we don't have to do that

Reply:
resolved offline